apps/api/src/controllers/competition.controller.ts

```diff
@@ -473,6 +473,161 @@ export function makeCompetitionController(services: ServiceRegistry) {
+            `[CompetitionController] Admin requesting competition details`,
+          );
+        } else if (agentId) {
+          console.log(
+            `[CompetitionController] Agent ${agentId} requesting competition details`,
+          );
+        } else if (userId) {
+          console.log(
+            `[CompetitionController] User ${userId} requesting competition details`,
+          );
+        } else {
+          throw new ApiError(401, "Authentication required");
+        }
+
+        // Get competition ID from path parameter
+        const competitionId = req.params.competitionId;
+        if (!competitionId) {
+          throw new ApiError(400, "Competition ID is required");
+        }
+
+        // Get competition details
+        const competition =
+          await services.competitionManager.getCompetition(competitionId);
+        if (!competition) {
+          throw new ApiError(404, "Competition not found");
+        }
+
+        // Return the competition details
+        res.status(200).json({
+          success: true,
+          competition: competition,
+        });
+      } catch (error) {
+        next(error);
+      }
+    },
+
+    /**
+     * Get agents participating in a competition
+     * @param req AuthenticatedRequest object with agent authentication information
+     * @param res Express response
+     * @param next Express next function
+     */
+    async getCompetitionAgents(
+      req: AuthenticatedRequest,
+      res: Response,
+      next: NextFunction,
+    ) {
+      try {
+        const agentId = req.agentId;
+        const userId = req.userId;
+        const isAdmin = req.isAdmin === true;
+
+        // Authentication check
+        if (isAdmin) {
+          console.log(
+            `[CompetitionController] Admin requesting competition agents`,
+          );
+        } else if (agentId) {
+          console.log(
+            `[CompetitionController] Agent ${agentId} requesting competition agents`,
+          );
+        } else if (userId) {
+          console.log(
+            `[CompetitionController] User ${userId} requesting competition agents`,
+          );
+        } else {
+          throw new ApiError(401, "Authentication required");
+        }
+
+        // Get competition ID from path parameter
+        const competitionId = req.params.competitionId;
+        if (!competitionId) {
+          throw new ApiError(400, "Competition ID is required");
+        }
+
+        // Check if competition exists
+        const competition =
+          await services.competitionManager.getCompetition(competitionId);
+        if (!competition) {
+          throw new ApiError(404, "Competition not found");
+        }
+
+        // Get leaderboard data for the competition
+        const leaderboard =
+          await services.competitionManager.getLeaderboard(competitionId);
+
+        // Get all agents to get their details
+        const agents = await services.agentManager.getAllAgents();
```

Comment:
Is this getting every agent in the database, or only the agents for the given competition?

Reply:
ah yeah, we should add a method for `getAgentForCompetition` or just have a `getAgents` method with an optional set of common where conditions (like `competitionId`)

Reply:
FWIW, this is what I did in https://github.com/recallnet/js-recall/pull/447
There's an example of filter, sort, etc.. here: https://github.com/recallnet/js-recall/pull/447/files#diff-8f37cbe3a6530858f8f9257279d174bbfce73923fe0187e82ba28cb090be72d0R167